dd-sdk-android-ndk/src/main/cpp/datadog-native-lib.cpp

```diff
@@ -34,11 +49,23 @@ std::string get_serialized_log(int signal,
                                const char *backtrace) {
     std::string serializedLog;
     serializedLog.append("{ ");
-    serializedLog.append(R"("message": ")").append(GENERIC_MESSAGE).append("\",");
-    serializedLog.append(R"("service": ")").append(SERVICE_NAME).append("\",");
-    serializedLog.append(R"("logger.name": ")").append(LOGGER_NAME).append("\",");
-    serializedLog.append(R"("env": ")").append(ENVIRONMENT).append("\",");
-    serializedLog.append(R"("status": ")").append(EMERGENCY_STATUS).append("\",");
+    if (rumcontext.application_id[0] != '\0') {
+        serializedLog.append(R"("application_id": ")").append(rumcontext.application_id)
+                .append("\",");
+    }
+    if (rumcontext.session_id[0] != '\0') {
+        serializedLog.append(R"("session_id": ")").append(rumcontext.session_id)
+                .append("\",");
+    }
+    if (rumcontext.view_id[0] != '\0') {
+        serializedLog.append(R"("view.id": ")").append(rumcontext.view_id)
+                .append("\",");
+    }
+    serializedLog.append(R"("message": ")").append(context.generic_message).append("\",");
+    serializedLog.append(R"("service": ")").append(context.service_name).append("\",");
+    serializedLog.append(R"("logger.name": ")").append(context.logger_name).append("\",");
+    serializedLog.append(R"("env": ")").append(context.environment).append("\",");
```

Comment:
`env` should be send as a tag, not log attribute:
```json
{
"service" : "default-service-name",
"logger.version": "1.2.2",
"ddtags": "env:tests"
}
```

Here's how we pass it inside `Logger.kt`: https://github.com/DataDog/dd-sdk-android/blob/4b2234f97c01629ac6fbbf03def62be45638f150/dd-sdk-android/src/main/kotlin/com/datadog/android/log/Logger.kt#L485

and [`LogSerializer.kt`](https://github.com/DataDog/dd-sdk-android/blob/master/dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/domain/LogSerializer.kt) doesn't work with the `env` tag directly.

Reply:
Good catch 👍